File: Backend/src/main/java/aqct/backend/model/Respuesta.java
package aqct.backend.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import java.io.Serializable;
import java.sql.Timestamp;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import org.hibernate.annotations.CreationTimestamp;

@Entity
@Table(name = "respuesta")
public class Respuesta implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private long id;

    @Column(name = "respuesta")
    private String respuesta;

    @Column(name = "aprobado")
    private boolean aprobado;

    @Column(name = "created_at")
    @CreationTimestamp
    private Timestamp createdAt;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "pregunta_id")
    @JsonBackReference
    private Pregunta pregunta;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "usuario_id")
    private Usuario usuario;
    
    public Respuesta() {
        
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getRespuesta() {
        return respuesta;
    }

    public void setRespuesta(String respuesta) {
        this.respuesta = respuesta;
    }

    public boolean isAprobado() {
        return aprobado;
    }

    public void setAprobado(boolean aprobado) {
        this.aprobado = aprobado;
    }

    public Timestamp getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(Timestamp createdAt) {
        this.createdAt = createdAt;
    }

    public Pregunta getPregunta() {
        return pregunta;
    }

    public void setPregunta(Pregunta pregunta) {
        this.pregunta = pregunta;
    }

    public Usuario getUsuario() {
        return usuario;
    }

    public void setUsuario(Usuario usuario) {
        this.usuario = usuario;
    }
    
}

File: aqct-aplication/src/model/respuesta.ts
import { Usuario } from "./usuario";
import { Pregunta } from "./pregunta";

export class Respuesta {

    public id: number;
    public createdAt: Date;
    public respuesta: string;
    public aprobado: boolean;

    public pregunta: Pregunta;
    public usuario: Usuario;

    public constructor(json?: any) {

        Object.assign(this, json);

        if (this.createdAt) { this.createdAt = new Date(this.createdAt) };
        if (this.pregunta) { this.pregunta = new Pregunta(this.pregunta) };
        if (this.usuario) { this.usuario = new Usuario(this.usuario) };

    }

    public getJSON(): any {
        return {
            id: this.id,
            createdAt: this.createdAt,
            respuesta: this.respuesta,
            aprobado: this.aprobado
        }
    }

}